src/test/java/hudson/plugins/copyartifact/CopyArtifactWorkflowTest.java

```diff
@@ -146,4 +156,173 @@ private WorkflowJob createWorkflow(String name, String script) throws IOExceptio
         job.setDefinition(new CpsFlowDefinition("node {" + script + "}", true));
         return job;
     }
+
+    // Tests for symbols.
+    // Test only that symbols are accessible as expected.
+    // No tests for features here as they should be tested in other tests.
+    // The latest pipeline supports calling `copyArtifacts` without `step`.
+
+    @Test
+    public void testDownstreamBuildSelector() throws Exception {
+        // upstream (freestyle) -> copiee (fresstyle)
+        // copier (pipeline) copies from copiee, which is downstream of `upstream`.
+        // DownstreamBuildSelector support detecting relations between only `AbstractProject`s.
+
+        FreeStyleProject upstream = jenkinsRule.createFreeStyleProject("upstream");
+        upstream.getBuildersList().add(new FileWriteBuilder("upstream_artifact.txt", "${BUILD_TAG}"));
+        ArtifactArchiver aa = new ArtifactArchiver("upstream_artifact.txt");
+        aa.setFingerprint(true);        // important to have Jenkins track builds
+        upstream.getPublishersList().add(aa);
+
+        FreeStyleProject copiee = jenkinsRule.createFreeStyleProject("copiee");
+        CopyArtifact ca = new CopyArtifact("upstream");
+        ca.setFingerprintArtifacts(true);       // important to have Jenkins track builds
+        copiee.getBuildersList().add(ca);
+        copiee.getBuildersList().add(new FileWriteBuilder("artifact.txt", "foobar"));
+        copiee.getPublishersList().add(new ArtifactArchiver("artifact.txt"));
+
+        jenkinsRule.assertBuildStatusSuccess(upstream.scheduleBuild2(0));
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0));
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: downstream(upstreamProjectName: 'upstream', upstreamBuildNumber: '1')));"
```

Comment:
Oh I see what you are saying—you did not in fact get the test to work properly without `step`. Meaning you need to update, IIRC, `workflow-basic-steps`.